bigquery_etl/operational_monitoring/templates/query.sql

```diff
@@ -0,0 +1,94 @@
+{{ header }}
+
+WITH merged_probes AS (
+  SELECT
+    @submission_date AS submission_date,
+    client_id,
+    SAFE.SUBSTR(application.build_id, 0, 8) AS build_id,
+    {% for dimension in dimensions %}
+      CAST({{ dimension.sql }} AS STRING) AS {{ dimension.name }},
+    {% endfor %}
+    mozfun.map.get_key(
+      environment.experiments,
+      "{{slug}}"
+    ).branch AS branch,
+    ARRAY<
+      STRUCT<
+        metric STRING,
+        histograms ARRAY<
+          STRUCT<
+            bucket_count INT64,
+            sum INT64,
+            histogram_type INT64,
+            `range` ARRAY<INT64>,
+            values ARRAY<STRUCT<key INT64, value INT64>>>
+        >>
+    >[
+      {% for probe in probes %}
+        (
+            "{{ probe.name }}",
+            ARRAY_AGG(mozfun.hist.extract({{ probe.sql }}) IGNORE NULLS)
+        )
+        {{ "," if not loop.last else "" }}
+      {% endfor %}
+    ] AS metrics,
+  FROM
+    `{{source}}`
+  WHERE
+    DATE(submission_timestamp) >= DATE_SUB(@submission_date, INTERVAL 30 DAY)
+  GROUP BY
+    1,
+    2,
+    3,
+    4,
+    5,
+    6
```

Comment:
It looks like `dimensions` could be an arbitrary set of dimensions. Could this break the `GROUP BY` here (e.g. if `dimensions` happen to contain 100 different dimensions)?

Reply:
oh good catch, thank you